src/app/components/home/home.component.ts

```diff
@@ -66,60 +66,64 @@ export class HomeComponent implements OnInit {
     this.sidenavService.emitChangeFromSidenav(candidateKey);
   }
 
-  massageCandidateData(){
-
-    this.modifiedData["city council"]={} as CandidateTree;
-    this.modifiedData["city council"]["title"]="City Council";
-    this.modifiedData["city council"]["candidates"]={};
-    
-    var candidateObj= this.candidates;
- 
-    const entries=Object.entries(this.candidates);
-
-    entries.forEach(entry=>{
-      if(!entry["0"].toLowerCase().includes("city-council")){
-        let test= entry["0"];
-        this.modifiedData[test]= {} as CandidateTree ;
-        this.modifiedData[entry["0"]]= entry["1"];
-      }else{
-        entry["1"]["name"]= entry["1"].title.slice(15);
-        let hyphenatedString=this.createTitle(entry);
-        entry["1"].title=  hyphenatedString.replace(/\s/g, '').toLowerCase().trim();
-       
-        this.modifiedData["city council"]["candidates"][entry["0"]]=entry["1"];
+  // data is turned into a key value pair
+  //object like this
+  //{
+  //  mayor:{},
+  //  city-council:{},
+  //  city-attorney:{}
+  //}
+  massageCandidateData() {
+
+    this.modifiedData["city council"] = {} as CandidateTree;
+    this.modifiedData["city council"]["title"] = "City Council";
+    this.modifiedData["city council"]["name"] = "City Council";
+    this.modifiedData["city council"]["deepTree"] = true;
+    this.modifiedData["city council"]["candidates"] = {};
+
+    const entries = Object.entries(this.candidates);
+
+    entries.forEach(entry => {
+      if (!entry["0"].toLowerCase().includes("last")) {
+        if (!entry["0"].toLowerCase().includes("city-council")) {
+          this.modifiedData[entry["0"]] = entry["1"];
+          this.modifiedData[entry["0"]]["deeptree"]= false;
+        } else {
+          this.modifiedData["city council"]["candidates"][entry["0"]] = entry["1"];
+        }
       }
 
     });
 
-    this.sortedObj=this.sortObj(this.modifiedData);
+    this.sortedObj = this.sortObj(this.modifiedData);
   }
 
-  sortObj(modifiedObject){
-    let temp={};
-    let modData= this.modifiedData;
-     var sortedEntries= Object.keys(modifiedObject).sort(function(a, b){
-         return b.charCodeAt(0)-a.charCodeAt(0);
+
+  private sortObj(modifiedObject) {
+    let temp = {};
+    let modData = this.modifiedData;
+    var sortedEntries = Object.keys(modifiedObject).sort(function (a, b) {
+      return b.charCodeAt(0) - a.charCodeAt(0);
     });
-   
-    sortedEntries.forEach(x=>{
-       temp[x]= modData[x];
+
+    sortedEntries.forEach(x => {
+      temp[x] = modData[x];
     });
 
     return temp;
-        
-  }
 
-  private createTitle(entry:[string, CandidateTree]){
-    let stringOne=entry["1"].title.replace(' ' ,'-').slice(0,22);
-    let stringTwo= entry["1"].title.slice(24);
-    let hyphenatedString= `${stringOne}-${stringTwo}`;
-    return hyphenatedString;
   }
-  otherOfficeClicked(office:string){
-    // 
-  }
-  
-  asIsOrder(a,b){
+
+
+  // private createTitle(entry: [string, CandidateTree]) {
```

Comment:
Should **createTitle** be removed if it is not being used?

Reply:
yea it can be removed.